pkg/cmd/start/start.go

```diff
@@ -525,6 +540,25 @@ func (o *Options) combineWithCustomParameters(params []job.PipelineRunParam) []j
 	return params
 }
 
+func (o *Options) addCustomEnvsToStepTemplate(spec *v1beta1.PipelineSpec) {
+	for name, value := range o.customEnvMap {
+		for i := range spec.Tasks {
+			stepTemplate := spec.Tasks[i].TaskSpec.StepTemplate
+			stepTemplate.Env = append(stepTemplate.Env, v1.EnvVar{
+				Name:  name,
+				Value: value,
+				ValueFrom: &v1.EnvVarSource{
```

Comment:
Remove `ValueFrom`

As I wrote as a response to your question there should be either a `Value` or `ValueFrom` in an `EnvVar`. Since the value is supplied verbatim it should be `Value`.

Reply:
My bad I hadn't pushed the changes.